drivers/adc/adc_stm32.c

```diff
@@ -236,6 +236,41 @@ static const uint32_t table_samp_time[] = {
 };
 #endif
 
+/* table to define the LL ADC clock prescaler from the DTS */
+static const uint32_t adc_clock_prescaler[] = {
+#if defined(CONFIG_SOC_SERIES_STM32F0X) || \
+	defined(CONFIG_SOC_SERIES_STM32L0X) || \
+	defined(CONFIG_SOC_SERIES_STM32WLX)
+	LL_ADC_CLOCK_SYNC_PCLK_DIV1,
+	LL_ADC_CLOCK_SYNC_PCLK_DIV2,
+	LL_ADC_CLOCK_SYNC_PCLK_DIV4,
+#elif defined(STM32F3X_ADC_V1_1) || \
+	defined(CONFIG_SOC_SERIES_STM32L4X) || \
+	defined(CONFIG_SOC_SERIES_STM32L5X) || \
+	defined(CONFIG_SOC_SERIES_STM32WBX) || \
+	defined(CONFIG_SOC_SERIES_STM32G0X) || \
+	defined(CONFIG_SOC_SERIES_STM32G4X) || \
+	defined(CONFIG_SOC_SERIES_STM32H7X)
+	LL_ADC_CLOCK_SYNC_PCLK_DIV1,
+	LL_ADC_CLOCK_SYNC_PCLK_DIV2,
+	LL_ADC_CLOCK_SYNC_PCLK_DIV4,
+#elif defined(CONFIG_SOC_SERIES_STM32L1X) || \
+	defined(CONFIG_SOC_SERIES_STM32U5X)
+	LL_ADC_CLOCK_ASYNC_DIV1,
+	LL_ADC_CLOCK_ASYNC_DIV2,
+	LL_ADC_CLOCK_ASYNC_DIV4,
+	LL_ADC_CLOCK_ASYNC_DIV6,
+	LL_ADC_CLOCK_ASYNC_DIV8,
+	LL_ADC_CLOCK_ASYNC_DIV10,
+	LL_ADC_CLOCK_ASYNC_DIV12,
+	LL_ADC_CLOCK_ASYNC_DIV16,
+	LL_ADC_CLOCK_ASYNC_DIV32,
+	LL_ADC_CLOCK_ASYNC_DIV64,
+	LL_ADC_CLOCK_ASYNC_DIV128,
+	LL_ADC_CLOCK_ASYNC_DIV256,
+#endif
```

Comment:
These are also supported by most other series: CONFIG_SOC_SERIES_STM32L4X || CONFIG_SOC_SERIES_STM32L5X || CONFIG_SOC_SERIES_STM32WBX || CONFIG_SOC_SERIES_STM32G0X || CONFIG_SOC_SERIES_STM32G4X || CONFIG_SOC_SERIES_STM32H7X

Reply:
CONFIG_SOC_SERIES_STM32U5X and CONFIG_SOC_SERIES_STM32L0X only have LL_ADC_CLOCK_ASYNC_DIVx  for their common clock config. 
Soc series that have LL_ADC_CLOCK_SYNC_PCLK_DIVx and also LL_ADC_CLOCK_ASYNC_DIVx to set their LL_ADC_SetCommonClock, would require a prescaler for Synchronous clock and one for Asynchronous clock.